pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -226,6 +230,34 @@ def restore_optimizers_and_schedulers(self) -> None:
         self.restore_optimizers()
         self.restore_lr_schedulers()
 
+    def restore_result_collections(self) -> None:
+        """ Restores the loop result collections used for logging."""
+        if not self._loaded_checkpoint:
+            return
+
+        state_dict = self._loaded_checkpoint.get('result_collections', None)
+        if state_dict:
+            # get current reduce function
+            sync_fn = self.trainer.training_type_plugin.reduce
+
+            # get current result collections
+            train_results = self.trainer.train_loop.results
+            val_results = self.trainer.evaluation_loop._val_results
+            test_results = self.trainer.evaluation_loop._test_results
+
+            should_reset = not self.trainer.is_global_zero
```

Comment:
why do this?

Reply:
Because we need to reset them on non rank 0 process as states were transferred already.